compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/CodeAnalyzer.java

```diff
@@ -839,9 +840,29 @@ public void visit(BLangArrayLiteral arrayLiteral) {
     }
 
     public void visit(BLangRecordLiteral recordLiteral) {
-        recordLiteral.keyValuePairs.forEach(kv -> {
+        List<BLangRecordLiteral.BLangRecordKeyValue> keyValuePairs = recordLiteral.keyValuePairs;
+        keyValuePairs.forEach(kv -> {
             analyzeExpr(kv.valueExpr);
         });
+
+        Set<Object> names = new TreeSet<>((l, r) -> l.equals(r) ? 0 : 1);
+        for (BLangRecordLiteral.BLangRecordKeyValue recFieldDecl : keyValuePairs) {
+            BLangExpression key = recFieldDecl.getKey();
+            if (key.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
+                BLangSimpleVarRef keyRef = (BLangSimpleVarRef) key;
+                if (names.contains(keyRef.variableName.value)) {
+                    String assigneeType = recordLiteral.parent.type.getKind().typeName();
+                    this.dlog.error(key.pos, DiagnosticCode.DUPLICATE_KEY_IN_RECORD_LITERAL, assigneeType, keyRef);
+                }
+                names.add(keyRef.variableName.value);
+            } else if (key.getKind() == NodeKind.LITERAL) {
+                BLangLiteral keyLiteral = (BLangLiteral) key;
+                if (names.contains(keyLiteral.value)) {
```

Comment:
Don't we need to add the key to the names set, in the success case?

I think it will not give an error If someone write as `{"key": "value-1", "key": "value-2"}`, where both keys are literals. Shall we add a test case too?

Reply:
Good catch, CodeAnalyzer updated + test case added.